MobileCenterCrashes/MobileCenterCrashes/Internals/MSErrorAttachmentLog.m

```diff
@@ -0,0 +1,132 @@
+                     contentType:(NSString *)contentType {
+  if ((self = [self init])) {
+
+    // Convert NSData to base64 string.
+    NSString *dataString = [data base64EncodedStringWithOptions:NSDataBase64EncodingEndLineWithCarriageReturn];
+    self = [self initWithFilename:filename attachmentString:dataString contentType:contentType];
+  }
+  return self;
+}
+
+- (instancetype)initWithFilename:(nullable NSString *)filename attachmentText:(NSString *)text {
+  if ((self = [self init])) {
+    self = [self initWithFilename:filename attachmentString:text contentType:kMSTextType];
+  }
+  return self;
+}
+
+- (NSMutableDictionary *)serializeToDictionary {
+  NSMutableDictionary *dict = [super serializeToDictionary];
+
+  // Fill in the dictionary.
+  if (self.attachmentId) {
+    dict[kMSId] = self.attachmentId;
+  }
+  if (self.errorId) {
+    dict[kMSErrorId] = self.errorId;
+  }
+  if (self.contentType) {
+    dict[kMSContentType] = self.contentType;
+  }
+  if (self.filename) {
+    dict[kMSFileName] = self.filename;
+  }
+  if (self.data) {
+    dict[kMSData] = self.data;
+  }
+  return dict;
+}
+
+- (BOOL)isEqual:(id)object {
+
+  // TODO: We should also check for parent equalty with `![super isEqual:object]` but isEqual is not implemented
+  // everywhere.
+  if (!object || ![object isKindOfClass:[MSErrorAttachmentLog class]])
+    return NO;
+  MSErrorAttachmentLog *attachment = (MSErrorAttachmentLog *)object;
+  return ((!self.attachmentId && !attachment.attachmentId) || [self.attachmentId isEqualToString:attachment.attachmentId]) &&
+         ((!self.errorId && !attachment.errorId) || [self.errorId isEqualToString:attachment.errorId]) &&
+         ((!self.contentType && !attachment.contentType) || [self.contentType isEqualToString:attachment.contentType]) &&
+         ((!self.filename && !attachment.filename) || [self.filename isEqualToString:attachment.filename]) &&
+         ((!self.data && !attachment.data) || [self.data isEqualToString:attachment.data]);
+}
+
+- (BOOL)isValid {
+  return [super isValid] && self.errorId && self.attachmentId && self.filename && self.data && self.contentType;
+}
+
+#pragma mark - NSCoding
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+  self = [super initWithCoder:coder];
+  if (self) {
+    _attachmentId = [coder decodeObjectForKey:kMSId];
+    _errorId = [coder decodeObjectForKey:kMSErrorId];
+    _contentType = [coder decodeObjectForKey:kMSContentType];
+    _filename = [coder decodeObjectForKey:kMSFileName];
+    _data = [coder decodeObjectForKey:kMSData];
+  }
+  return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder {
+  [super encodeWithCoder:coder];
+  [coder encodeObject:self.attachmentId forKey:kMSId];
+  [coder encodeObject:self.errorId forKey:kMSErrorId];
+  [coder encodeObject:self.contentType forKey:kMSContentType];
+  [coder encodeObject:self.filename forKey:kMSFileName];
+  [coder encodeObject:self.data forKey:kMSData];
+}
+
+#pragma mark - Public Interface
```

Comment:
These should be in a separate class. These are utility methods, shouldn't be in *Log classes, we don't expose those internal classes. Android is using `ErrorAttachments` class, iOS can use `MSErrorAttachments` with static methods.